_arduino/nimbleProportionalControl.ino

```diff
@@ -0,0 +1,391 @@
+// ▝▘ ▝▘ ▝▀▘   ▀▀  ▝▀▘  ▀    ▀▀▀
+// ----------------------------------------------------------------
+
+// Motor globals
+int MIN_PWM_VALUE = 0; 
+
+int motorRightPWMValue = 0; 
+int motorLeftPWMValue = 0;
+
+int motorRightSpeedPercent;
+int motorLeftSpeedPercent;
+
+int uLeft;
+int uRight;
+
+MotorController motorRightController(HIGH, MOTOR_RIGHT_DIR, MOTOR_RIGHT_PWM);
+MotorController motorLeftController(LOW, MOTOR_LEFT_DIR, MOTOR_LEFT_PWM);
+
+Chrono motorControlChrono;
+
+// ----------------------------------------------------------------
+// ▗▄▄▖ ▗▄▖                                ▗▖   
+// ▐▛▀▜▌▝▜▌             ▐▌             ▐▌  ▐▌   
+// ▐▌ ▐▌ ▐▌  ▐▌ ▐▌ ▟█▙ ▐███  ▟█▙  ▟█▙ ▐███ ▐▙██▖
+// ▐███  ▐▌  ▐▌ ▐▌▐▙▄▟▌ ▐▌  ▐▛ ▜▌▐▛ ▜▌ ▐▌  ▐▛ ▐▌
+// ▐▌ ▐▌ ▐▌  ▐▌ ▐▌▐▛▀▀▘ ▐▌  ▐▌ ▐▌▐▌ ▐▌ ▐▌  ▐▌ ▐▌
+// ▐▙▄▟▌ ▐▙▄ ▐▙▄█▌▝█▄▄▌ ▐▙▄ ▝█▄█▘▝█▄█▘ ▐▙▄ ▐▌ ▐▌
+// ▝▀▀▀   ▀▀  ▀▀▝▘ ▝▀▀   ▀▀  ▝▀▘  ▝▀▘   ▀▀ ▝▘ ▝▘
+// ----------------------------------------------------------------
+
+#define SERVICE_UUID "19b10000-e8f2-537e-4f6c-d104768a1214"
+#define MEASURED_SPEED_CHARACTERISTIC_RIGHT_UUID "19b10001-e8f2-537e-4f6c-d104768a1214"
+#define MEASURED_SPEED_CHARACTERISTIC_LEFT_UUID "19b10002-e8f2-537e-4f6c-d104768a1214"
+#define SPEED_CHARACTERISTIC_RIGHT_UUID "19b10003-e8f2-537e-4f6c-d104768a1214"
+#define SPEED_CHARACTERISTIC_LEFT_UUID "19b10004-e8f2-537e-4f6c-d104768a1214"
+
+// Bluetooth globals
+
+static NimBLEServer* pServer;
+
+NimBLECharacteristic* pMeasuredSpeCharacteristicRight = NULL;
+NimBLECharacteristic* pMeasuredSpeedCharacteristicLeft = NULL;
+
+NimBLECharacteristic* pWebSpeedCharacteristicRight = NULL;
+NimBLECharacteristic* pWebSpeedCharacteristicLeft = NULL;
+
+bool deviceConnected = false;
+bool oldDeviceConnected = false;
+
+// TODO: take a look at this conversion and data type
+class SpeedCharacteristicCallbacks : public NimBLECharacteristicCallbacks {
+  void onWrite(NimBLECharacteristic* pSpeedCharacteristic) {
+    float speed = (pSpeedCharacteristic->getValue())[0];  // Get the value of the speed characteristic as a flloat
+     if (speed > 155) {
+        speed = speed - 256;
+      }
+      if (isRightMotorCallback_) {
+        motorRightSpeedPercent = speed;
+      } else {
+        motorLeftSpeedPercent = speed;
+      }  
+  }
+public:
+  bool isRightMotorCallback_;
+  SpeedCharacteristicCallbacks(bool right)
+    : isRightMotorCallback_(right), NimBLECharacteristicCallbacks() {}
+};
+
+class ServerCallbacks: public NimBLEServerCallbacks {
+    void onConnect(NimBLEServer* pServer) {
+        deviceConnected = true;
+    };
+    void onDisconnect(NimBLEServer* pServer) {
+        deviceConnected = false;
+        NimBLEDevice::startAdvertising();
+    };
+};
+
+void playTone(int tone, int duration) {
+  for (long i = 0; i < duration * 1000L; i += tone * 2) {
+    digitalWrite(speakerPin, HIGH);
+    delayMicroseconds(tone);
+    digitalWrite(speakerPin, LOW);
+    delayMicroseconds(tone);
+  }
+}
+
+void playNote(char note, int duration) {
```

Comment:
Does playing a note work?

Reply:
Works occasionally, but may be having PIN issues.